parsers/JP_ISEP.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+import os
+import re
+import json
+import logging
+import requests
+import datetime
+
+import pandas as pd
+
+url = 'https://isep-energychart.com/en/graphics/electricityproduction/?region={region}&period_year={year}&period_month={month}&period_day={day}&period_length=1day&display_format=residual_demand'
+timezone = 'Japan'
+
+MAP_ZONE_TO_REGION_NAME = {
+    'JP': 'all',
+    'JP-HKD': 'hokkaido',
+    'JP-TH': 'tohoku',
+    'JP-TK': 'tokyo',
+    'JP-CB': 'chubu',
+    'JP-HR': 'hokuriku',
+    'JP-KN': 'kansai',
+    'JP-CG': 'chugoku',
+    'JP-SK': 'shikoku',
+    'JP-KY': 'kyushu',
+    'JP-ON': 'okinawa'
+}
+
+COLUMN_MAP = {
+   'demand':'consumption',
+   'wind_performance':'wind',
+   'solar_performance':'solar',
+   'thermal':'unknown',
+   'pumped':'pumped hydro',
+   'interconnection':'exchanges',
+   'solar_suppression': 'solar curtailment',
+   'wind_suppression': 'wind curtailment'
+}
+
+def get_data(region, year, month, day):
+    r = requests.get(url.format(**{'region':region,
+                                    'year':year,
+                                    'month':month,
+                                    'day':day}))
```

Comment:
I'm not sure if you know - this can be done a little shorter without having to go via the dict:

```suggestion
    r = requests.get(url.format(region=region,
                                year=year,
                                month=month,
                                day=day))
```

Reply:
Didn't know that, thanks!